File: gto-support-jsonapi/src/main/java/org/ccci/gto/android/common/jsonapi/annotation/JsonApiPlaceholder.java
package org.ccci.gto.android.common.jsonapi.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * This annotation indicates this field will reflect whether the deserialized object is a placeholder object only.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface JsonApiPlaceholder {}

File: gto-support-jsonapi/src/main/java/org/ccci/gto/android/common/jsonapi/annotation/JsonApiPlaceholder.kt
package org.ccci.gto.android.common.jsonapi.annotation

/**
 * This annotation indicates this field will reflect whether the deserialized object is a placeholder object only.
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class JsonApiPlaceholder
